module/zfs/dmu.c

```diff
@@ -588,7 +588,9 @@ dmu_buf_rele_array(dmu_buf_t **dbp_fake, int numbufs, void *tag)
  * the data starting at offset, and continuing to offset + len.
  *
  * Note that if the indirect blocks above the blocks being prefetched are not in
- * cache, they will be asychronously read in.
+ * cache, they will be asychronously read in. Also, callers should make sure
+ * not to attempt to prefetch encrypted data without the relevant encryption key
+ * being loaded.
```

Comment:
Right now the zio code will start reporting EACCES errors if the objset is not owned when raw reads happen. The only time this becomes an issue is with zvols which try to prefetch sectors for `blkid` after creating the minor, but before it is mounted. I have added some special handling there.

I can change this comment to say `If the objset is encrypted, callers must ensure that the relevant key is loaded into the keystore or else the zio layer will report EACCES errors when attempting to decrypt the prefetched data.`